http_api.go

```diff
@@ -0,0 +1,194 @@
+package worker
+
+import (
+	"crypto/subtle"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+// APIHandler handles requests to worker's HTTP API.
+type APIHandler struct {
+	i *CLI
+}
+
+// Setup installs the HTTP routes that will handle requests to the HTTP API.
+func (api *APIHandler) Setup() {
+	api.i.logger.Info("setting up HTTP API")
+	r := mux.NewRouter()
+
+	r.HandleFunc("/healthz", api.HealthCheck).Methods("GET")
+
+	r.HandleFunc("/worker", api.GetWorkerInfo).Methods("GET")
+	r.HandleFunc("/worker", api.UpdateWorkerInfo).Methods("PATCH")
+	r.HandleFunc("/worker", api.ShutdownWorker).Methods("DELETE")
+
+	// It is preferable to use UpdateWorkerInfo to update the pool size,
+	// as it does not depend on the current state of worker.
+	r.HandleFunc("/pool/increment", api.IncrementPool).Methods("POST")
+	r.HandleFunc("/pool/decrement", api.DecrementPool).Methods("POST")
+
+	r.Use(api.CheckAuth)
+	http.Handle("/", r)
+}
+
+// CheckAuth is a middleware for all HTTP API methods that ensures that the
+// configured basic auth credentials were passed in the request.
+func (api *APIHandler) CheckAuth(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// skip auth for the health check endpoint
+		if strings.HasPrefix(req.URL.Path, "/healthz") {
+			next.ServeHTTP(w, req)
+			return
+		}
+
+		username, password, ok := req.BasicAuth()
+		if !ok {
+			w.Header().Set("WWW-Authenticate", "Basic realm=\"travis-ci/worker\"")
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+
+		authBytes := []byte(fmt.Sprintf("%s:%s", username, password))
+		if subtle.ConstantTimeCompare(authBytes, []byte(api.i.c.String("http-api-auth"))) != 1 {
```

Comment:
This is the only place where `api.i.c` is accessed, so I think I would prefer it be injected as a string value instead of reaching through `.i.c.String`.